### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by tgalery**
Connects to #47

### CODE REVIEW

File: src/main/java/it/cnr/isti/hpc/wikipedia/reader/WikipediaArticleReader.java [hunk 1]
```diff
@@ -154,22 +156,26 @@ public void process(WikiArticle page, Siteinfo si) {
 				return;
 			}
 
-			if(page.isHelp()){
+			Boolean isHelp = integerNamespace.equals(NamespaceCode.HELP_NAMESPACE_KEY.code);
```
Comment (hmcc) [inline src/main/java/it/cnr/isti/hpc/wikipedia/reader/WikipediaArticleReader.java]: Boolean with a small `b`; Java [Object.equals](https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#equals-java.lang.Object-) only returns true or false, never null.

Comment (hmcc) [reply src/main/java/it/cnr/isti/hpc/wikipedia/reader/WikipediaArticleReader.java]: Actually, might be neater to just do away with the whole thing and do switch-case:
```
switch(integerNamespace) {
    case NamespaceCode.HELP_NAMESPACE_KEY.code:
        // stuff
        break;
    ...
}
```

**Comment by hmcc** [comment]
Connecting issue?

File: src/main/java/it/cnr/isti/hpc/wikipedia/reader/WikipediaArticleReader.java (hunk 1 above)
Comment (tgalery) [reply src/main/java/it/cnr/isti/hpc/wikipedia/reader/WikipediaArticleReader.java]: Good spot @hmcc

Comment (stathischaritos) [reply src/main/java/it/cnr/isti/hpc/wikipedia/reader/WikipediaArticleReader.java]: switch case did look nicer but apparently I can't cause they're not constants :(

**Comment by tgalery** [comment]
@hmcc I've updated the connecting issue, sorry for missing it ;-)
Moreover, @stathischaritos and I used this branch to create a new `RunWikistatsPairCounts` on the `simple` wikipedia. Comparing that to an old task that has been stored in the remote cache, we get the following:

```bash
diff --git a/newPairCountsSorted.txt b/oldPairCountsSorted.txt
index 7f1ae66..0dc2e7c 100644
--- a/newPairCountsSorted.txt
+++ b/oldPairCountsSorted.txt
@@ -8960,3 +8960,2 @@ AFI's 100 Years... 100 Laughs     http://dbpedia.org/resource/AFI's_100_Years..._100
 AFI's 100 Years... 100 Movies  http://dbpedia.org/resource/AFI's_100_Years..._100_Movies       14
-AFI's 100 Years...100 Movies   http://dbpedia.org/resource/AFI's_100_Years..._100_Movies       14
 AFI's 100 Years... 100 Stars   http://dbpedia.org/resource/AFI's_100_Years..._100_Stars        3
@@ -50389,4 +50388,6 @@ crumple zone    http://dbpedia.org/resource/Crumple_zone        1
 Crumple zone   http://dbpedia.org/resource/Crumple_zone        1
+Crunk & B      http://dbpedia.org/resource/Crunk       1
 Crunk&B        http://dbpedia.org/resource/Crunk       1
 Crunk & B      http://dbpedia.org/resource/Crunk_&_B   1
+Crunk&B        http://dbpedia.org/resource/Crunk_&_B   1
 Crunk  http://dbpedia.org/resource/Crunk       1
@@ -59546,4 +59547,4 @@ DuckTales       http://dbpedia.org/resource/DuckTales   22
 DuckTales the Movie: Treasure of the Lost Lamp http://dbpedia.org/resource/DuckTales_the_Movie:_Treasure_of_the_Lost_Lamp      6
-Ducktales: Treasure of the Golden Suns http://dbpedia.org/resource/Ducktales:_Treasure_of_the_Golden_Suns      1
-Ducktales:Treasure of the Golden Suns  http://dbpedia.org/resource/Ducktales:_Treasure_of_the_Golden_Suns      1
+Ducktales: Treasure of the Golden Suns http://dbpedia.org/resource/Ducktales:_Treasure_of_the_Golden_Suns      2
+Ducktales:Treasure of the Golden Suns  http://dbpedia.org/resource/Ducktales:_Treasure_of_the_Golden_Suns      2
 Duck typing    http://dbpedia.org/resource/Duck_typing 1
@@ -68570,3 +68571,2 @@ father  http://dbpedia.org/resource/William_II_of_the_Netherlands       1
 father- in-law http://dbpedia.org/resource/Father      1
-father-in-law  http://dbpedia.org/resource/Father      1
 Father Junipero Serra  http://dbpedia.org/resource/Junipero_Serra      1
@@ -72767,3 +72767,2 @@ Fort Lauderdale, Florida        http://dbpedia.org/resource/Fort_Lauderdale_–_Hollywo
 Fort Lauderdale – Hollywood International Airport      http://dbpedia.org/resource/Fort_Lauderdale_–_Hollywood_International_Airport   6
-Fort Lauderdale–Hollywood International Airport        http://dbpedia.org/resource/Fort_Lauderdale_–_Hollywood_International_Airport   6
 Fort Lauderdale        http://dbpedia.org/resource/Fort_Lauderdale,_Florida    19
@@ -105648,2 +105647,3 @@ Kingdom Hearts III    http://dbpedia.org/resource/Kingdom_Hearts_III  3
 Kingdom Hearts: Re:Chain of Memories   http://dbpedia.org/resource/Kingdom_Hearts:_Chain_of_Memories   1
+Kingdom Hearts Re: Chain of Memories   http://dbpedia.org/resource/Kingdom_Hearts:_Chain_of_Memories   5
 Kingdom Hearts Re:Chain of Memories    http://dbpedia.org/resource/Kingdom_Hearts:_Chain_of_Memories   5
@@ -109942,2 +109942,3 @@ Law and Order SVU     http://dbpedia.org/resource/Law_&_Order:_Special_Victims_Unit
 Law and Order: SVU     http://dbpedia.org/resource/Law_&_Order:_Special_Victims_Unit   1
+Law and Order:SVU      http://dbpedia.org/resource/Law_&_Order:_Special_Victims_Unit   1
 La Wantzenau   http://dbpedia.org/resource/La_Wantzenau        4
@@ -110927,2 +110928,3 @@ Legolas       http://dbpedia.org/resource/Middle-earth_characters     10
 Lego Ninjago: Masters of Spinjitzu     http://dbpedia.org/resource/Lego_Ninjago:_Masters_of_Spinjitzu  2
+Lego Ninjago:Masters of Spinjitzu      http://dbpedia.org/resource/Lego_Ninjago:_Masters_of_Spinjitzu  2
 Lego Star Wars 3: The Clone Wars       http://dbpedia.org/resource/Lego        1
@@ -124026,5 +124028,5 @@ Mexican-American      http://dbpedia.org/resource/Mexican–American_War        1
 Mexican American War   http://dbpedia.org/resource/Mexican–American_War        1
+Mexican- American War  http://dbpedia.org/resource/Mexican–American_War        1
 Mexican–American War   http://dbpedia.org/resource/Mexican–American_War        28
-Mexican- American War  http://dbpedia.org/resource/Mexican–American_War        60
-Mexican-American War   http://dbpedia.org/resource/Mexican–American_War        60
+Mexican-American War   http://dbpedia.org/resource/Mexican–American_War        59
 Mexican Association of Phonograph Producers    http://dbpedia.org/resource/AMPROFON    1
@@ -135052,2 +135054,3 @@ Ninigi        http://dbpedia.org/resource/Ninigi-no-Mikoto    1
 Ninigi-no-Mikoto       http://dbpedia.org/resource/Ninigi-no-Mikoto    4
+Ninjago: Masters of Spinjitzu  http://dbpedia.org/resource/Lego_Ninjago:_Masters_of_Spinjitzu  1
 Ninjago:Masters of Spinjitzu   http://dbpedia.org/resource/Lego_Ninjago:_Masters_of_Spinjitzu  1
@@ -166818,3 +166821,2 @@ San Francisco International Airport   http://dbpedia.org/resource/San_Francisco_In
 San Francisco Municipal Railway        http://dbpedia.org/resource/San_Francisco_Municipal_Railway     1
-San Francisco – Oakland Bay Bridge     http://dbpedia.org/resource/San_Francisco–Oakland_Bay_Bridge    3
 San Francisco–Oakland Bay Bridge       http://dbpedia.org/resource/San_Francisco–Oakland_Bay_Bridge    3
@@ -179776,2 +179778,3 @@ Star Trek: The Motion Picture http://dbpedia.org/resource/Star_Trek:_The_Motion_
 Star Trek: The Next Generation http://dbpedia.org/resource/Star_Trek:_The_Next_Generation      53
+Star Trek:The Next Generation  http://dbpedia.org/resource/Star_Trek:_The_Next_Generation      53
 Star Trek: The Original Series http://dbpedia.org/resource/Star_Trek:_The_Original_Series      15
@@ -194956,4 +194959,4 @@ Trent Green   http://dbpedia.org/resource/Trent_Green 3
 Trent  http://dbpedia.org/resource/River_Trent 4
-Trentino - Alto Adige  http://dbpedia.org/resource/Trentino-Alto_Adige/Südtirol        3
-Trentino-Alto Adige    http://dbpedia.org/resource/Trentino-Alto_Adige/Südtirol        3
+Trentino - Alto Adige  http://dbpedia.org/resource/Trentino-Alto_Adige/Südtirol        1
+Trentino-Alto Adige    http://dbpedia.org/resource/Trentino-Alto_Adige/Südtirol        2
 Trentino-Alto Adige/Südtirol   http://dbpedia.org/resource/Trentino-Alto_Adige/Südtirol        28
```

To be honest I think these differences come from the tokenizer in Babel (the old file is pre babel) and the changes are not that much. But this does seem to motive some token chain normalisation function in babel to be used in wikistats before agreegating sfs. Yeah ?

File: src/main/java/it/cnr/isti/hpc/wikipedia/reader/WikipediaArticleReader.java [hunk 2]
```diff
@@ -154,22 +156,19 @@ public void process(WikiArticle page, Siteinfo si) {
 				return;
 			}
 
-			if(page.isHelp()){
+			if(integerNamespace.equals(NamespaceCode.HELP_NAMESPACE_KEY.code)){
```
Comment (Lugrin) [inline src/main/java/it/cnr/isti/hpc/wikipedia/reader/WikipediaArticleReader.java]: Can we delete the `WikiArticle.isHelp` method then? Same question for changes below.

Comment (Lugrin) [reply src/main/java/it/cnr/isti/hpc/wikipedia/reader/WikipediaArticleReader.java]: @stathischaritos said this method is gone in the new version of bliki and that's precisely the reason of the change above.

**Comment by tgalery** [comment]
fyi, created this one https://github.com/idio/babel/issues/13 to capture those cases.

**Pull request merged**